marklogic-data-hub/src/main/java/com/marklogic/hub/impl/MasteringManagerImpl.java

```diff
@@ -128,4 +145,47 @@ public MergeResponse merge(List<String> mergeURIs, String flowName, String stepN
             return resp;
         }
     }
+
+    static class MatchResource extends ResourceManager {
+        private String targetDatabase;
+
+        public MatchResource(DatabaseClient srcClient, String targetDatabase) {
+            super();
+            this.targetDatabase = targetDatabase;
+            srcClient.init("mlSmMatch" , this);
+        }
+
+        public MatchResponse match(String matchURI, String flowName, String stepNumber, Boolean includeMatchDetails, JsonNode options) {
+            MatchResponse resp;
+
+            RequestParameters params = new RequestParameters();
+            params.put("uri", matchURI);
+            params.put("includeMatchDetails", includeMatchDetails.toString());
+            params.put("flowName", flowName);
+            params.put("step", stepNumber);
+            params.put("targetDatabase", targetDatabase);
+            params.put("sourceDatabase", targetDatabase);
+            JacksonHandle jsonOptions = new JacksonHandle().with(options);
+            ResourceServices.ServiceResultIterator resultItr = this.getServices().post(params, jsonOptions);
+            try {
+                if (resultItr == null || !resultItr.hasNext()) {
+                    resp = null;
+                } else {
+                    ResourceServices.ServiceResult res = resultItr.next();
+                    StringHandle handle = new StringHandle();
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+                    objectMapper.enable(DeserializationFeature.UNWRAP_ROOT_VALUE);
```

Comment:
Does using a JacksonHandle to capture the response not work because these features need to be set? i.e. does this not work?

    return getServices().post(params, jsonOptions, new JacksonHandle()).get();

That would give you a JsonNode instead of a MatchResponse, but I'm wondering if MatchResponse is really needed. Who's the client for this interface? If it's a GUI, odds are the MatchResponse class isn't needed and a JsonNode will work.

Reply:
Good point. Refactored to use more generic JacksonHandle/JsonNode.